### PR TITLE
Split ConfigurationProperties in two: auth and deployment conf.

### PR DESCRIPTION
**Pull request by ericbottard**
Also, add infrastructure so that the latter can be independently provided
for apps and tasks (with the default being to share a single instance).

Fixes spring-cloud/spring-cloud-deployer-cloudfoundry#66

### CODE REVIEW

File: src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerTests.java [hunk 1]
```diff
@@ -137,21 +140,22 @@ public void shouldNotNamespaceTheDeploymentIdWhenNoGroupIsUsed() throws Interrup
...
 		final String barKey = "another.cloud.bar";
 		final String barVal = "this should too";
 
-		deploymentProperties.put(fooKey, fooVal);
-		deploymentProperties.put(barKey, barVal);
+		appDeploymentProperties.put(fooKey, fooVal);
+		appDeploymentProperties.put(barKey, barVal);
 
-		deployer = new CloudFoundryAppDeployer(properties, operations, client, deploymentCustomizer);
+		deployer = new CloudFoundryAppDeployer(properties, deploymentProperties, operations, client, deploymentCustomizer);
```
Comment (ilayaperumalg) [inline src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerTests.java]: should this be `appDeploymentProperties` here?

Comment (ericbottard) [inline src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerTests.java]: No, I believe this is correct (check the types). `appDeploymentProperties ` are passed later below.

**Comment by ilayaperumalg** [comment]
I see the following two test failures on this branch:

```
 testSimpleCancel(org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryTaskLauncherIntegrationTests): Source emitted more than one item
  cleanUp(org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryTaskLauncherIntegrationTests): CF-BadQueryParameter(10005): The query parameter is invalid: Unknown query parameter(s): 'app_guids'
```

**Comment by ericbottard** [comment]
Did you export appropriate ENV VARS, something like
```
export SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_SERVICES=rabbit
export SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_TASK_SERVICES=mysql
```

**Comment by markfisher** [comment]
@ericbottard: your last comment got my attention. We need to avoid any awareness of "stream" at the level of the deployer (only dataflow should know about streams). The env var you posted above is binding to the nested `stream()` as exposed by the deployment properties and used here: https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/pull/67/files#diff-7a16c7ef5ac913f477a4cb15870fa1e1R213

**Comment by ericbottard** [comment]
@markfisher @ilayaperumalg Actually, my comment above is totally wrong. Those vars would have no effect at the *deployer* level. They would only work at the server level, thanks to https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/pull/152/files#diff-f0ede45fb30e46b2ea916d21b3c93c7e

The ENV VAR you'd need to test this code would simply use `CLOUDFOUNDRY_PROPERTIES` as the prefix. But you shouldn't even require it.

**Comment by markfisher** [comment]
@ericbottard but this is still a violation right? https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/pull/67/files#diff-7a16c7ef5ac913f477a4cb15870fa1e1R214

**Comment by ericbottard** [comment]
@markfisher I'm not sure what you mean (neither do I know which line 214 I should look at). None of these keys have `.stream.` in their name, have they?

**Comment by markfisher** [comment]
@ericbottard This is what I'm referring to: ```deploymentProperties.getServices().stream()```

**Comment by ericbottard** [comment]
Ah. Did not see that. But, behold, it's https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#stream-- :)

**Comment by markfisher** [comment]
Haha. I was reading that as a nested property of the ConfigurationProperties class... nevermind then ;)

**Comment by ilayaperumalg** [comment]
LGTM, verified the splitting of properties as well. Created issue https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/issues/70 for task launching.

Merging.

**Comment by ilayaperumalg** [comment]
Squashed and merged as 11e93f3

**Pull request closed without merging**